tests/org.eclipse.e4.ui.tests.css.core/src/org/eclipse/e4/ui/tests/css/core/parser/MediaRulesTest.java

```diff
@@ -29,18 +30,22 @@
 public class MediaRulesTest {
 
 	@Test
-	@Disabled("//THIS TEST KNOWN TO FAIL Dec 16/08")
 	void testMediaRule() throws Exception {
 		String css = """
-			@media screen, print {
-			BODY { line-height: 1.2 }
-			}
-			Label { background-color: #FF0000 }""";
+				@media screen, print {
+					BODY { line-height: 1.2 }
+					Label { background-color: #FFFFFF }
+				}
+				BODY { line-height: 1.3 }
+				Label { background-color: #FF0000 }""";
 		CSSStyleSheet styleSheet = ParserTestUtil.parseCss(css);
 		assertNotNull(styleSheet);
-		CSSRuleList rules = styleSheet.getCssRules();
-
+		List<String> cssRules = IntStream.range(0, styleSheet.getCssRules().getLength())
+				.mapToObj(i -> styleSheet.getCssRules().item(i).getCssText()).toList();
 
-		assertEquals(1, rules.getLength());
+		assertThat(cssRules.stream().filter(it -> it.contains("background-color")).toList())
+		.containsOnly("Label { background-color: rgb(255, 0, 0); }");
+		assertThat(cssRules.stream().filter(it -> it.contains("line-height")).toList())
+				.containsOnly("BODY { line-height: 1.3; }");
 	}
```

Comment:
```suggestion
	CSSStyleSheet styleSheet = ParserTestUtil.parseCss(css);
	assertNotNull(styleSheet);

	// This one is provided only inside the @media so it shouldn't be there
	assertThat(findCssRuleThatContains(styleSheet, "background-color")).isEmpty();

	// This one is provided outside the @media and it shouldn't be overwritten by
	// any of them
	assertThat(findCssRuleThatContains(styleSheet, "line-height")).containsOnly("BODY { line-height: 1.3; }");
}

private Stream<String> findCssRuleThatContains(CSSStyleSheet styleSheet, String text) {
	Stream<String> cssRulesText = IntStream.range(0, styleSheet.getCssRules().getLength())
			.mapToObj(i -> styleSheet.getCssRules().item(i).getCssText());
	return cssRulesText.filter(r -> r.contains(text));
}
```

I extracted a private method to improve readability.